cloud-operator/templates/role.yaml

```
@@ -21,4 +21,15 @@ rules:
 - apiGroups: [""]
   resources: ["secrets"]
   resourceNames: ["hubble-relay-client-certs"] # Name of the Secret
-  verbs: ["get"]
\ No newline at end of file
+  verbs: ["get"]
+---
+# Role to read secrets in the GKE managed observability namespace
+apiVersion: rbac.authorization.k8s.io/v1
+kind: Role
+metadata:
+  name: illumio-cloud-operator-secret-reader
+  namespace: gke-managed-dpv2-observability
+rules:
+- apiGroups: [""]
+  resources: ["secrets"]
```

The secret name is hardcoded ([here](https://github.com/illumio/cloud-operator/pull/258/files#diff-6d4dd1d6018369bf594f74c982493806edfd65dbc3a000f9157924d645baddbcR28)), so you should limit the role to access only this one resource, like in the Role for namespace `kube-system` above:
```suggestion
  resources: ["secrets"]
  resourceNames: ["hubble-relay-client-certs"] # Name of the Secret
```
Both Roles should be completely identical except for the namespace.